Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserBus.h

```diff
@@ -128,6 +128,39 @@ namespace AzToolsFramework
 
         typedef AZStd::vector<SourceFileOpenerDetails> SourceFileOpenerList;
 
+        //! This struct is used to respond about being able to create source files.
+        //! See AssetBrowserInteractionNotifications::OpenSourceFileInEditor below
+        struct SourceFileCreatorDetails
+        {
+            //! You provide a function to call (you may use std-bind to bind to your class) if your creator is chosen to handle the create
+            //! operation
+            typedef AZStd::function<void(const char* /*fullSourceFileName*/, const AZ::Uuid& /*source uuid*/)> SourceFileCreatorFunctionType;
+
+            AZStd::string m_identifier; ///< choose something unique for your opener.  It may be used to restore state.  it will not be shown to user.
+
+            //! m_displayText is used when more than one listener offers to create this kind of file and
+            //! we need the user to pick which one they want.  They will be offered all the available creators in a menu
+            //! which shows this text, and the one they pick will get its SourceFileCreatorFunctionType called.
+            AZStd::string m_displayText;
+            QIcon m_iconToUse; ///< optional.  Same as m_displayText.  Used when there's ambiguity.  If empty, no icon.
+
+            //! This is the function to call.  If you fill a nullptr in here, then the default operating system behavior will be suppressed
+            //! but no creator will be opened.  This will also cause the 'create' option in context menus to disappear if the only creators
+            //! are nullptr ones.
+            SourceFileCreatorFunctionType m_creator;
+
+            SourceFileCreatorDetails() = default;
+            SourceFileCreatorDetails(const char* identifier, const char* displayText, QIcon icon, SourceFileCreatorFunctionType functionToCall)
+                : m_identifier(identifier)
+                , m_displayText(displayText)
+                , m_iconToUse(icon)
+                , m_creator(functionToCall)
+            {
+            }
```

Comment:
`Actually it would be better to accept `AZStd::string` by value and `AZStd::move` into the member variables, not by const reference as that would require two copies.

Reply:
Changed to AZStd::string and using AZStd::move.